File: JULY-2025/26-07-2025/p3480.py
class Solution:
    def maxSubarrays(self, n: int, conflictingPairs: List[List[int]]) -> int:
        valid = 0

        conflictingpoints = [[] for _ in range(n+1)]
        for i in conflictingPairs:
            conflictingpoints[max(i)].append(min(i))

        conflict = 0
        second_conflict = 0

        extra = [0]*(n+1)

        for end in range(1, n+1):  #will count total subarrays ending at this point 
            #check all conflicting points end
            for u in conflictingpoints[end]:
                if u>=conflict:
                    second_conflict = conflict
                    conflict = u
                elif u > second_conflict:
                    second_conflict = u
            valid+=(end - conflict)
            extra[conflict]+=(conflict-second_conflict)

        return valid + max(extra)

File: JULY-2025/26-07-2025/p3480.cpp
class Solution {
public:
    long long maxSubarrays(int n, vector<vector<int>>& conflictingPairs) {
        long long valid = 0;
        vector<vector<int>> conflictingPoints(n + 1);
        for (auto& p : conflictingPairs) {
            int a = min(p[0], p[1]), b = max(p[0], p[1]);
            conflictingPoints[b].push_back(a); 
        }
        int maxConflictPoint = 0;
        int secondMaxConflictPoint = 0;
        vector<long long> extra(n + 1, 0);
        for (int end = 1; end <= n; end++) {
            for (auto& u : conflictingPoints[end]) {
                if (u >= maxConflictPoint) {
                    secondMaxConflictPoint = maxConflictPoint;
                    maxConflictPoint = u;
                } else if (u > secondMaxConflictPoint)
                    secondMaxConflictPoint = u;
            }
            valid += end - maxConflictPoint;
            extra[maxConflictPoint] +=
                maxConflictPoint - secondMaxConflictPoint;
        }
        return valid + *max_element(extra.begin(), extra.end());
    }
};